packages/eservice-descriptors-archiver/src/services/eserviceDescriptorsArchiver.ts

```diff
@@ -0,0 +1,67 @@
+import { Logger, RefreshableInteropToken } from "pagopa-interop-commons";
+import { Agreement, DescriptorId, EServiceId } from "pagopa-interop-models";
+import { v4 as uuidv4 } from "uuid";
+import { CatalogProcessClient } from "./catalogProcessClient.js";
+import { ReadModelService } from "./readModelService.js";
+import { archiveDescriptorForArchivedAgreement } from "./archiveDescriptorProcessor.js";
+
+export type EServiceDescriptorsArchiver = {
+  archiveDescriptorsForArchivedAgreement: (
+    archivedAgreement: Agreement,
+    logger: Logger,
+    correlationId: string | undefined | null
+  ) => Promise<void>;
+};
+
+export const eserviceDescriptorArchiverBuilder = async (
+  refreshableToken: RefreshableInteropToken,
+  readModelService: ReadModelService,
+  catalogProcessClient: CatalogProcessClient
+): Promise<EServiceDescriptorsArchiver> => {
+  // eslint-disable-next-line @typescript-eslint/explicit-function-return-type
+  const getHeaders = (
+    correlationId: string | undefined | null,
+    token: string
+  ) => ({
+    "X-Correlation-Id": correlationId || uuidv4(),
+    Authorization: `Bearer ${token}`,
+  });
+
+  const archiveDescriptor = async (
+    descriptorId: DescriptorId,
+    eserviceId: EServiceId,
+    correlationId: string | undefined | null
+  ): Promise<void> => {
+    const token = (await refreshableToken.get()).serialized;
+    const headers = getHeaders(correlationId, token);
+
+    await catalogProcessClient.archiveDescriptor(undefined, {
+      params: {
+        eServiceId: eserviceId,
+        descriptorId,
+      },
+      headers,
+    });
+  };
+
+  return {
+    archiveDescriptorsForArchivedAgreement: async (
+      archivedAgreement: Agreement,
+      logger: Logger,
+      correlationId: string | undefined | null
+    ): Promise<void> => {
+      const archivedDescriptor = await archiveDescriptorForArchivedAgreement(
+        archivedAgreement,
+        readModelService,
+        archiveDescriptor,
+        logger,
+        correlationId
+      );
+      if (archivedDescriptor) {
+        logger.info(
+          `Sent Descriptor archive request for archived Agreement ${archivedAgreement.id} - Descriptor ${archivedAgreement.descriptorId} - EService ${archivedAgreement.eserviceId}`
```

Comment:
from the message it seems that the request is async, which is a bit misleading, because if we arrive here the descriptor has already been archived.

If the suggestion to remove this file will be accepted, we could move the confirmation log [here](https://github.com/pagopa/interop-be-monorepo/pull/465/files?diff=split&w=1#diff-29d32aef94752c0597104440c9179bef036ae6852c3d478c5ffb222188ebbc02R69) and [here](https://github.com/pagopa/interop-be-monorepo/pull/465/files?diff=split&w=1#diff-29d32aef94752c0597104440c9179bef036ae6852c3d478c5ffb222188ebbc02R84)

Reply:
Done and moved logs where the call to catalog process `/archive` actually happens